libbeat/filter/registry.go

```diff
@@ -0,0 +1,29 @@
+package filter
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type FilterRule interface {
+	Filter(event common.MapStr) (common.MapStr, error)
+	CheckConfig(c common.Config) error
```

Comment:
You are right, I am removing it as it should be the decision of the filter plugin if it wants to do any checking.